src/PSRule.Rules.Azure/rules/Azure.VM.Rule.yaml

```
@@ -110,6 +110,24 @@ spec:
     field: properties.dnsSettings.dnsServers
     hasValue: false
 
+---
+# Synopsis: VMs should be deallocated instead of stopped.
+apiVersion: github.com/microsoft/PSRule/v1
+kind: Rule
+metadata:
+  name: Azure.VM.ShouldNotBeStopped
+  ref: AZR-123346
+  tags:
+    release: 'GA'
+    ruleSet: '2020_06'
+    Azure.WAF/pillar: Operational Excellence
+spec:
+  type:
+  - Microsoft.Compute/virtualMachines
+  condition:
```

Let's add a selector to only apply to in-flight cases, otherwise this rule may trigger for infrastructure as code which will never has tis value set.

```suggestion
  with:
  - PSRule.Rules.Azure\Azure.Resource.Export
  condition:
```